notebooks/3.jump-analysis/3.jump_analysis.py

```diff
@@ -0,0 +1,270 @@
+    strict=True
+)
+shuffled_multi_class_model_path = (
+    modeling_dir / "shuffled_multi_class_model.joblib"
+).resolve(strict=True)
+feature_col_names = (data_split_dir / "feature_cols.json").resolve(strict=True)
+injury_codes_path = (data_split_dir / "injury_codes.json").resolve(strict=True)
+
+# output paths
+jump_analysis_dir = (results_dir / "3.jump_analysis").resolve()
+jump_analysis_dir.mkdir(exist_ok=True)
+
+
+# ## loading files
+
+# In[3]:
+
+
+# loading in the negatlive controled normalized profiles
+jump_df = pd.read_csv(jump_data_path)
+
+# loading json file containing selected feature names
+with open(feature_col_names, mode="r") as infile:
+    cell_injury_cp_feature_cols = json.load(infile)
+
+# loading json file that contains the coder and decoder injury labels
+with open(injury_codes_path) as infile:
+    injury_codes = json.load(infile)
+
+injury_decoder = injury_codes["decoder"]
+injury_encoder = injury_codes["encoder"]
+
+# display dataframe and size
+print("JUMP dataset size:", jump_df.shape)
+jump_df.head()
+
+
+# ## Feature alignment
+# In this section, we are identifying the shared features present in both the cell injury and JUMP datasets.
+# Once these features are identified, we update the JUMP dataset to include only those features that are shared between both profiles for our machine learning application.
+
+# First we identify the CellProfiler (CP) features present in the JUMP data.
+# We accomplish this by utilizing `pycytominer`'s  'infer_cp_features()', which helps us identify CP features in the JUMP dataset.
+
+# In[4]:
+
+
+# get compartments
+metadata_prefix = "Metadata_"
+compartments = list(
+    set(
+        [
+            feature_name.split("_")[0]
+            for feature_name in jump_df.columns.tolist()
+            if not feature_name.startswith("Metadata_")
+        ]
+    )
+)
+
+# find CP features in JUMP dataset
+jump_cp_features = infer_cp_features(jump_df, compartments=compartments)
+meta_features = infer_cp_features(jump_df, compartments=compartments, metadata=True)
+
+# display number of features of both profiles
+print("Number of Metadata Features:", len(meta_features))
+print(
+    "Number of CP features that cell injury has",
+    len(cell_injury_cp_feature_cols["feature_cols"]),
+)
+print("Number of CP features that JUMP has:", len(jump_cp_features))
+
+
+# Now that we have identified the features present in both datasets, the next step is to align them. This involves identifying the common features between both profiles and utilizing these features to update our JUMP dataset for our machine learning model.
+
+# In[5]:
+
+
+cell_injury_cp_features = cell_injury_cp_feature_cols["feature_cols"]
+
+# finding shared featues using intersection
+aligned_features = list(set(cell_injury_cp_features) & set(jump_cp_features))
+
+# displaying number of shared features between both profiles
+print("Number of shapred features of both profiles", len(aligned_features))
```

Comment:
```suggestion
print("Number of shared features of both profiles", len(aligned_features))
```

Also, wow! There are only 207 features in common? Do you know the source of the discrepancy? This is also a really small proportion... we might be able to solve the discrepancy by understanding the differences (maybe there are small differences in naming that we can solve), or, we might need to retrain the models using the smaller subset.